src/transformers/trainer.py

```diff
@@ -2969,7 +2969,20 @@ def _save_rng_state(self, output_dir):
     def _save_optimizer_and_scheduler(self, output_dir):
         if is_torch_xla_available():
             xm.rendezvous("saving_optimizer_states")
-            xm.save(self.optimizer.state_dict(), os.path.join(output_dir, OPTIMIZER_NAME))
+            if self.is_fsdp_xla_enabled and not self.is_fsdp_xla_v2_enabled:
```

Comment:
What does it mean if `self.is_fsdp_xla_enabled` is True and `self.is_fsdp_xla_v2_enabled` is False? Is it possible to have all possible combinations of these flags? e.g.: 

```
self.is_fsdp_xla_enabled=True, self.is_fsdp_xla_v2_enabled=False
self.is_fsdp_xla_enabled=True, self.is_fsdp_xla_v2_enabled=True
self.is_fsdp_xla_enabled=False, self.is_fsdp_xla_v2_enabled=False
self.is_fsdp_xla_enabled=False, self.is_fsdp_xla_v2_enabled=True
```

As this check is repeated several times below - it's be good to combine into a single, explicitly named flag e.g. `fsdp_v1_enabled`

Reply:
Good suggestion, I have already changed to `self.is_fsdp_xla_v1_enabled`. `self.is_fsdp_xla_v1_enabled` and `self.is_fsdp_xla_v2_enabled` are different versions of the fsdp xla implementation.